### PR TITLE
Add float16 and bfloat16 support and test for argsort

### PR DESCRIPTION
**Pull request by enkilee**
### PR types

Others
### PR changes

Others
### Description

* https://github.com/PaddlePaddle/Paddle/issues/54871 - 2 argsort
Add float16 and bfloat16 support and test for argsort

### CODE REVIEW

**Comment by enkilee** [comment]
@zhangting2020 婷姐，下面这个报错，怎么会到mean_grad那去了？

```bash
2023-07-07 19:05:43 NotFoundError: The kernel with key (GPU, Undefined(AnyLayout), int64) of kernel `mean_grad` is not registered and fail to fallback to CPU one. Selected wrong DataType `int64`. Paddle support following DataTypes: complex64, bool, float16, bfloat16, complex128, float32, float64.
2023-07-07 19:05:43   [Hint: Expected kernel_iter != iter->second.end(), but received kernel_iter == iter->second.end().] (at ..\paddle\phi\core\kernel_factory.cc:304)
```

**Comment by zhangting2020** [comment]
> @zhangting2020 婷姐，下面这个报错，怎么会到mean_grad那去了？
> 
> ```shell
> 2023-07-07 19:05:43 NotFoundError: The kernel with key (GPU, Undefined(AnyLayout), int64) of kernel `mean_grad` is not registered and fail to fallback to CPU one. Selected wrong DataType `int64`. Paddle support following DataTypes: complex64, bool, float16, bfloat16, complex128, float32, float64.
> 2023-07-07 19:05:43   [Hint: Expected kernel_iter != iter->second.end(), but received kernel_iter == iter->second.end().] (at ..\paddle\phi\core\kernel_factory.cc:304)
> ```

我调试看看原因

File: test/legacy_test/test_argsort_op.py [hunk 1]
```diff
@@ -513,5 +514,95 @@ def test_fp16(self):
...
+        self.axis = 0
+
+    def init_direction(self):
+        self.descending = False
+
+    def test_check_output(self):
+        self.check_output(no_check_set=['Indices'])
+
+    def test_check_grad(self):
+        self.check_grad(['X'], 'Out', no_grad_set=['Indices'])
```
Comment (zhangting2020) [inline test/legacy_test/test_argsort_op.py]: 单测系统存在一处bug，后面我会修复，ci报错问题，这里先通过如下方式解决，bf16的case也一样

self.check_grad_with_place(place, ['X'], 'Out', check_dygraph=False)

File: test/legacy_test/test_argsort_op.py [hunk 2]
```diff
@@ -513,5 +514,95 @@ def test_fp16(self):
...
+        self.input_shape = [
+            10000,
+        ]
+        self.axis = 0
+
+    def init_direction(self):
+        self.descending = False
+
+    def test_check_output(self):
+        self.check_output(no_check_set=['Indices'])
```
Comment (zhangting2020) [inline test/legacy_test/test_argsort_op.py]: 这里应该不用设置no_check_set

File: test/white_list/no_check_set_white_list.py [hunk 3]
```diff
@@ -39,4 +39,5 @@
     'rmsprop',
     'rrelu',
     'layer_norm',
+    'argsort',
```
Comment (zhangting2020) [inline test/white_list/no_check_set_white_list.py]: 白名单的修改要去掉

File: test/white_list/no_grad_set_white_list.py [hunk 4]
```diff
@@ -70,4 +70,5 @@
     'smooth_l1_loss',
     'spectral_norm',
     'complex',
+    'argsort',
```
Comment (zhangting2020) [inline test/white_list/no_grad_set_white_list.py]: 这里也不要加

File: test/legacy_test/test_argsort_op.py (hunk 1 above)
Comment (enkilee) [reply test/legacy_test/test_argsort_op.py]: 收到

File: test/white_list/no_check_set_white_list.py (hunk 3 above)
Comment (enkilee) [reply test/white_list/no_check_set_white_list.py]: 收到

File: test/legacy_test/test_argsort_op.py (hunk 2 above)
Comment (enkilee) [reply test/legacy_test/test_argsort_op.py]: 收到

File: test/white_list/no_grad_set_white_list.py (hunk 4 above)
Comment (enkilee) [reply test/white_list/no_grad_set_white_list.py]: 收到

**Review by zhangting2020 — APPROVED** [review]
记得更新中文文档

**Pull request merged**